source/kubernetes/versions.rst

```diff
@@ -0,0 +1,116 @@
+.. _kubernetes-versions:
+
+########
+Versions
+########
+
+*******************
+Kubernetes Versions
+*******************
+
+Kubernetes community releases a minor version about very three months. In those
+minor version releases, there are some new features and bug fixes. Patch versions
+will be released more frequent (e.g. weekly) and genereally to include critical
+fixes , e.g. security fixes.
+
+Catalyst Cloud Kubernetes Service supports each ``minor`` version for at least
+``6`` months before deprecating it to give users enough time to upgrade their
```

Comment:
I'd like to get you guys comments about the versions support policies. Because I find it is not an easy task. Generally, there are about 10 (more or less) patch versions for each minor version. They're published per 3-4 weeks. So the whole period for one minor version may take 1 year. And I have seen some cloud providers support that long, e.g. IBM and EKS.

The challenging part is the patch version really. Now we generally only maintain one patch version for each minor version, that said, the old patch version will be removed as long as there is a new patch version. For example, if the current minor version is v.1.16.8 and the v1.16.9 just released. Now we just hide the v1.16.8 in favor of v.1.16.9.
This is OK actually, we can still say we support each minor version for at least 6 months. It's easy to achieve. However, the tricky thing is, should we say the cluster still on v1.16.8 is out of support if it's not upgraded in 30 days.

Reply:
There are some questions we need to discuss since I don't have a good answer yet:
1. Are we going to provide every patch version? Or we can start from some patch we believe it's stable for that minor version.
2. Personally, I'd like to support 1 year for each minor version. And at any time, we support two patch versions as I proposed below. 

However, assuming we release a new patch version per month, e.g. v1.16.8, after about 2 months, there could be a new patch version v1.16.10, and we're going to deprecate/remove the v1.16.8 patch version. And we give users 30 days to upgrade. For this case, that means for about every 3 months, users have to upgrade their clusters to a new patch version. Do you think that's acceptable?

Reply:
Looks great Feilong. 

To be clear is we say we support each version for 6 months before deprecating it, is the 30 days you refer to after that period? If so then I feel that is a fair and reasonable limit to set.

Reply:
I guess the only time the 30 days period might be too short a time frame is where we introduce a breaking change that does not support the rolling upgrade functionality e.g. the recent change of OS for the cluster nodes. In a case like this we may want to consider a slightly longer window perhaps?

With regards to the including every patch version I like your idea of going with the stable patch version and I feel that if we rolled them up and did it every 3 months then they (the customer) could retain the version they are on for a period of around 6 months before needing to upgrade in order to stay on a supported version.

Reply:
Regarding the breaking change point above, this versioning scheme does not express that at all. Can that be surfaced somehow, either in the versioning scheme or somewhere else, so that it's obvious to the user when they are on version N but cannot upgrade N+1 to the next due to one of these "gaps"?

Reply:
> I guess the only time the 30 days period might be too short a time frame is where we introduce a breaking change that does not support the rolling upgrade functionality e.g. the recent change of OS for the cluster nodes. In a case like this we may want to consider a slightly longer window perhaps?
> 
> With regards to the including every patch version I like your idea of going with the stable patch version and I feel that if we rolled them up and did it every 3 months then they (the customer) could retain the version they are on for a period of around 6 months before needing to upgrade in order to stay on a supported version.

The Fedora Atomic -> Fedora CoreOS is a special case, it won't happen again after GA.